File: backend/ai/prompts/triage.py
from . import SAFETY_PREAMBLE

TRIAGE_SYSTEM_PROMPT = SAFETY_PREAMBLE + (
    "\n\nYou perform preliminary symptom triage. Respond with ONLY a JSON object, no markdown "
    "fences, no commentary, matching exactly this shape: "
    '{"likely_causes": [{"condition": str, "probability": "low"|"medium"|"high", "explanation": str}, ...], '
    '"urgency_level": "home_care"|"see_doctor_soon"|"emergency", "next_steps": str}. '
    "List two to four likely causes ordered from most to least probable. Be conservative: if there is "
    "any doubt or any red-flag combination of symptoms, prefer a higher urgency level."
)


def build_triage_prompt(intake_data: dict, health_context: str) -> str:
    parts = [
        f"Main symptom or concern: {intake_data.get('main_symptom', 'not specified')}",
        f"Onset: {intake_data.get('onset', 'not specified')}",
        f"Duration: {intake_data.get('duration', 'not specified')}",
        f"Severity: {intake_data.get('severity', 'not specified')}",
        f"Associated symptoms: {intake_data.get('associated_symptoms', 'none reported')}",
    ]
    if health_context:
        parts.append(f"Relevant health history: {health_context}")
    return "\n".join(parts)

File: frontend/src/api/triage.ts
import { apiClient } from "./client";

export type UrgencyLevel = "home_care" | "see_doctor_soon" | "emergency";

export interface LikelyCause {
  condition: string;
  probability: "low" | "medium" | "high";
  explanation: string;
}

export interface TriageResult {
  id: number;
  likely_causes: LikelyCause[];
  urgency_level: UrgencyLevel;
  next_steps: string;
  ai_provider_used: string;
  is_mock: boolean;
  created_at: string;
}

export interface TriageIntake {
  main_symptom: string;
  onset: string;
  duration: string;
  severity: string;
  associated_symptoms: string;
}

export interface TriageSession {
  id: number;
  intake_data: TriageIntake;
  result: TriageResult | null;
  created_at: string;
}

export async function createTriageSession(intake_data: TriageIntake) {
  const { data } = await apiClient.post<TriageSession>("/triage/sessions/", { intake_data });
  return data;
}

export async function listTriageSessions() {
  const { data } = await apiClient.get<{ results: TriageSession[] } | TriageSession[]>("/triage/sessions/");
  return Array.isArray(data) ? data : data.results;
}
